### PR TITLE
Fixed rerun logic and exit handling in first-boot.

### PR DESCRIPTION
**Pull request by illfelder**

### CODE REVIEW

File: google-startup-scripts/usr/share/google/first-boot [hunk 1]
```diff
@@ -32,41 +31,28 @@ function get_instance_id() {
...
-  return $?
-}
+declare -r INSTANCE_ID=$(get_instance_id)
 
-sync_new_instance_check
-if [[ $? != 0 ]]; then
-  # The instance hasn't changed.  Just exit.
-  exit 0
-fi
+# Checks whether the instance has changed.
```
Comment (null0000) [inline google-startup-scripts/usr/share/google/first-boot]: Might help to add a note here that the function exits if the instance has already been booted.

**Comment by null0000** [comment]
LGTM

File: google-startup-scripts/usr/share/google/first-boot (hunk 1 above)
Comment (illfelder) [inline google-startup-scripts/usr/share/google/first-boot]: Done.

**Comment by zmarano** [comment]
LGTM

**Pull request merged**